mbuild/examples/reverse_mapping/mapping_moieties/propane_aa.py

```diff
@@ -0,0 +1,16 @@
+import mbuild as mb
+
+class Propane_aa(mb.Compound):
```

Comment:
Pedantic comment: this is propyl. Is there a reason that you could not simply express this using the existing `Alkane` example without capping one end? Would reduce the file clutter a bit I think

Reply:
I'm not sure what the python terminology is called, but I tried use a dictionary of python modules(?) like in https://github.com/ahy3nz/mbuild/blob/reverse_mapping/mbuild/compound.py#L1571, but applying it to 'mapping moieties' in https://github.com/ahy3nz/mbuild/blob/reverse_mapping/mbuild/coarse_graining.py#L50. If I used the `Alkane` class, I'd have to pass certain args to make the single-un-capped propyl. I think passing these args isn't 'transparent' to the user in the example https://github.com/ahy3nz/mbuild/blob/reverse_mapping/mbuild/examples/reverse_mapping/revmap_hexane_cg.py#L11. 
I do believe your suggestion is the more flexible/re-usable option, but I just don't know the best way to implement it

Reply:
Ah I see - that does complicate things a little. Technically you can achieve this effect via [`functools.partial`](https://docs.python.org/2/library/functools.html#functools.partial) but that's getting a bit complicated for this relatively simple purpose.

The "dictionary of python modules" is simply a dict where the vals are functions (functions are first class objects in Python meaning you can pass them around as objects without evaluating them). 

Is there a reason why you couldn't pass instances of these compounds instead of the class reference [here](https://github.com/mosdef-hub/mbuild/pull/394/files#diff-5873106fe9d5fedc4e9abf3c98f4e5d5R50)? Then you easily pass any fully evaluated compound into the function instead of evaluating them inside the function which would solve this particular problem. Not sure if there are any broader ramifications.